examples/mpi/mpibench/test.bats

```diff
@@ -83,18 +80,22 @@ check_process_ct () {
 }
 
 # This test compares OpenMPI's point to point bandwidth with all high speed
-# plugins enabled against the performance just using tcp. If the performance of
-# the former is not at least double that of tcp the former's performance is
-# assumed to be lacking.
+# plugins enabled against the performance just using TCP. Pass if HSN
+# performance is at least double TCP.
 @test "${ch_tag}/using the high-speed network (host launch)" {
     multiprocess_ok
-    [[ $SLURM_NNODES = "1" ]] && skip "Multinode only"
+    [[ $ch_multinode ]] || skip "multinode only"
     [[ $ch_cray ]] && skip "Cray doesn't support running on tcp"
+    # Verify we have proper HSN devices present. (Note that -d tests for
+    # directory, not device.)
+    [[ -d /dev/infiniband ]]
```

Comment:
This will cause the test to fail on Ethernet only clusters e.g. woodchuck, do we plan to not test on such systems?

Reply:
In principle an Ethernet system could be using RDMA, though we do not have any such systems.

Our Ethernet clusters use TCP, making the test a no-op. We'd want to skip in that case, I think.

Reply:
Apologies if I'm confused, in this PR it would fail on the [[ -d /dev/infiniband ]] check and in  master after the first unnecessary run of the benchmark it skips in disable_hsn_openmpi(). Which case would it fail because performance would be equal?

Reply:
You're right. It would also fail in performance if we modified the line in question to let Ethernet systems through.

Reply:
Ahh I see what you are saying, I wasn't suggesting removing the line, I was suggesting adding the skip back if the directory didn't exist. I wanted to see if this was intended behavior before suggesting it though.

Reply:
The reason I changed it from skip to fail is that I wanted to make an explicit choice about systems that have an HSN but it's not IB or OPA. But I can't think of a way to test whether there is an HSN present but it's Ethernet, since testing for Ethernet will match anything. Do you have any ideas?

Reply:
Makes sense. On the initial commit I only thought as far as RoCE which I believe also uses the device file /dev/infiniband/uverbsX so it would have passed the check for /dev/infiniband. My knowledge of Ethernet only system is quite lacking though so I don't have any ideas. If you have some example systems in mind I can look into it though.